### DISCUSSION TITLE
It is possible to get the name/icon of connected wallet using vue3 and ethers?

### DISCUSSION THREAD

**Discussion by zed-wong** [WalletConnect/walletconnect-monorepo#3921] [Developer Support]
I'm using web3modal v3 and found there's a wagmi solution, but it doesn't work for vue, is there a way to accomplish that using vue?

**Comment by glitch-txs** [comment]
this is not possible at the moment, I'll create an issue and see how we can solve it.

**Reply by glitch-txs** [reply]
you can keep track here: https://github.com/WalletConnect/web3modal/issues/1489

**Reply by zed-wong** [reply]
Okay, I also found an issue, using web3modal v3, when connecting from mobile, the connect success event wasn't triggered when connect success.

**Reply by glitch-txs** [reply]
which event? can you share your code?

**Reply by zed-wong** [reply]
```
  // Handle Web3Modal events
  const events = useWeb3ModalEvents();
  watchEffect(async () => {
    switch (events.data.event) {
      case "CONNECT_SUCCESS":
        const { walletProvider } = useWeb3ModalSigner()
        const { address } = useWeb3ModalAccount();
        console.log(walletProvider.value)
        cStore.afterConnect();
        console.log(address);
        appendConnected({
          name: WalletConnectName,
          icon: wc,
          chain: EthereumChainName,
          chain_id: ETHUUID,
          address: address.value || '',
        })
        break;
      case "DISCONNECT_SUCCESS":
        cStore.afterDisconnect();
        break;
      default:
        console.log(events.data.event);
    }
  });
```

`CONNECT_SUCCESS` event, I connected to my metamask mobile and the following code wasn't triggered.

**Reply by zed-wong** [reply]
https://github.com/WalletConnect/walletconnect-monorepo/assets/36271125/78649edf-7c02-4afe-89e0-08d0ae9c0c3e

On desktop, once the metamask is connected, the ui would change and it would be shown as connected, but in mobile it wasn't triggered. I guess that's because the event wasn't triggered?
